support/mesh/mesh_protobuf/src/lib.rs

```diff
@@ -596,48 +666,146 @@ mod tests {
         println!("{t:?}");
         let v = encode(t.clone());
         println!("{v:x?}");
+        expect.assert_eq(&as_expect_str(&v));
         let t2 = decode::<T>(&v).unwrap();
         assert_eq!(t, t2);
     }
 
     #[track_caller]
-    fn assert_field_roundtrips<T>(t: T)
+    fn assert_field_roundtrips<T>(t: T, expect: Expect)
     where
         T: crate::DefaultEncoding + Clone + Eq + core::fmt::Debug,
         T::Encoding: FieldEncode<T, NoResources> + for<'a> FieldDecode<'a, T, NoResources>,
     {
-        assert_roundtrips((t,));
+        assert_roundtrips((t,), expect);
     }
 
     #[test]
     fn test_field() {
-        assert_field_roundtrips(5u32);
-        assert_field_roundtrips(true);
-        assert_field_roundtrips("hi".to_string());
-        assert_field_roundtrips(5u128);
-        assert_field_roundtrips(());
-        assert_field_roundtrips((1, 2));
-        assert_field_roundtrips(("foo".to_string(), "bar".to_string()));
-        assert_field_roundtrips([1, 2, 3]);
-        assert_field_roundtrips(["abc".to_string(), "def".to_string()]);
-        assert_field_roundtrips(Some(5));
-        assert_field_roundtrips(Option::<u32>::None);
-        assert_field_roundtrips(vec![1, 2, 3]);
-        assert_field_roundtrips(vec!["abc".to_string(), "def".to_string()]);
-        assert_field_roundtrips(Some(Some(true)));
-        assert_field_roundtrips(Some(Option::<bool>::None));
-        assert_field_roundtrips(vec![None, Some(true), None]);
+        assert_field_roundtrips(
+            5u32,
+            expect!([r#"
+                1: varint 5
+                raw: 0805"#]),
+        );
+        assert_field_roundtrips(
+            true,
+            expect!([r#"
+                1: varint 1
+                raw: 0801"#]),
+        );
+        assert_field_roundtrips(
+            "hi".to_string(),
+            expect!([r#"
```

Comment:
Thing of beauty.

Reply:
![NiceGIF](https://github.com/user-attachments/assets/0d8b6f13-6312-4a54-aa50-5e2c94f8ee2e)